thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -1402,7 +1399,7 @@ public String deleteTrigger(
      */
     @NonNull
     @WorkerThread
-    public TargetState getTargetState(
+    public <S extends TargetState> S getTargetState(
```

Comment:
If S is not registered, throw `ClassCastException`. Mentioned it in API document.